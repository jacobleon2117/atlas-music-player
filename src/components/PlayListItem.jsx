export default function PlayListitem() {
  return (
    <div className="flex h-[44px] w-full items-center overflow-hidden">
      <div className="flex h-full flex-grow flex-col justify-between py-[1px]">
        <div className="flex h-[22px] items-center">
          <span className="truncate text-base font-medium">Electric Fever</span>
        </div>
        <div className="flex h-[22px] items-center">
          <span className="text-playlist-text-color truncate text-sm font-medium">
            Neon Jungle
          </span>
        </div>
      </div>
      <div className="flex h-full items-center">
        <span className="text-playlist-text-color text-sm">8:41</span>
      </div>
    </div>
  );
}
