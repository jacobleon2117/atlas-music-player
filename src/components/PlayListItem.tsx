interface PlayListItemProps {
  song: {
    id: string;
    title: string;
    artist: string;
    duration: number;
  };
  isSelected: boolean;
  onSelect: () => void;
}

export default function PlayListItem({ song, isSelected, onSelect }: PlayListItemProps) {
  const formatDuration = (seconds: number): string => {
    const mins = Math.floor(seconds / 60);
    const secs = seconds % 60;
    return `${mins}:${secs.toString().padStart(2, '0')}`;
  };

  return (
    <div
      onClick={onSelect}
      className={`flex h-[44px] cursor-pointer items-center rounded-lg px-2 transition-colors ${
        isSelected
          ? "bg-primary-light dark:bg-primary-sage/30 border-primary-sage mx-[-2px] border dark:border-white"
          : "hover:bg-primary-light/50 dark:hover:bg-primary-sage/20"
      }`}
    >
      <div className="flex-grow">
        <div className="text-primary-dark text-base font-medium dark:text-white">
          {song.title}
        </div>
        <div className="text-primary-sage dark:text-primary-sage/80 text-sm font-medium">
          {song.artist}
        </div>
      </div>
      <span className="text-primary-sage dark:text-primary-sage/80 text-sm font-medium">
        {formatDuration(song.duration)}
      </span>
    </div>
  );
}