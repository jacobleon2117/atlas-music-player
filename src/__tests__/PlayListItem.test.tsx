import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import PlayListItem from '../components/PlayListItem';

describe('PlayListItem', () => {
  const mockSong = {
    id: 'test-id',
    title: 'Test Song',
    artist: 'Test Artist',
    duration: 180
  };

  it('renders selected state', () => {
    const { container } = render(
      <PlayListItem
        song={mockSong}
        isSelected={true}
        onSelect={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders unselected state', () => {
    const { container } = render(
      <PlayListItem
        song={mockSong}
        isSelected={false}
        onSelect={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders with different duration formats', () => {
    const songWithLongDuration = {
      ...mockSong,
      duration: 3600
    };
    const { container } = render(
      <PlayListItem
        song={songWithLongDuration}
        isSelected={false}
        onSelect={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });
});