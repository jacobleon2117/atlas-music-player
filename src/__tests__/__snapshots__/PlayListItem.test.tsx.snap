// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem > renders selected state 1`] = `
<div>
  <div
    class="flex h-[44px] cursor-pointer items-center rounded-lg px-2 transition-colors bg-primary-light dark:bg-primary-sage/30 border-primary-sage mx-[-2px] border dark:border-white"
  >
    <div
      class="flex-grow"
    >
      <div
        class="text-primary-dark text-base font-medium dark:text-white"
      >
        Test Song
      </div>
      <div
        class="text-primary-sage dark:text-primary-sage/80 text-sm font-medium"
      >
        Test Artist
      </div>
    </div>
    <span
      class="text-primary-sage dark:text-primary-sage/80 text-sm font-medium"
    >
      3:00
    </span>
  </div>
</div>
`;

exports[`PlayListItem > renders unselected state 1`] = `
<div>
  <div
    class="flex h-[44px] cursor-pointer items-center rounded-lg px-2 transition-colors hover:bg-primary-light/50 dark:hover:bg-primary-sage/20"
  >
    <div
      class="flex-grow"
    >
      <div
        class="text-primary-dark text-base font-medium dark:text-white"
      >
        Test Song
      </div>
      <div
        class="text-primary-sage dark:text-primary-sage/80 text-sm font-medium"
      >
        Test Artist
      </div>
    </div>
    <span
      class="text-primary-sage dark:text-primary-sage/80 text-sm font-medium"
    >
      3:00
    </span>
  </div>
</div>
`;

exports[`PlayListItem > renders with different duration formats 1`] = `
<div>
  <div
    class="flex h-[44px] cursor-pointer items-center rounded-lg px-2 transition-colors hover:bg-primary-light/50 dark:hover:bg-primary-sage/20"
  >
    <div
      class="flex-grow"
    >
      <div
        class="text-primary-dark text-base font-medium dark:text-white"
      >
        Test Song
      </div>
      <div
        class="text-primary-sage dark:text-primary-sage/80 text-sm font-medium"
      >
        Test Artist
      </div>
    </div>
    <span
      class="text-primary-sage dark:text-primary-sage/80 text-sm font-medium"
    >
      60:00
    </span>
  </div>
</div>
`;
